docs/site/Relations.md

```diff
@@ -0,0 +1,164 @@
+the `has many` relation which operates by constraining the target repository by
+the foreign key property on its associated model.
+
+## Defining a HasMany Relation
+
+Model relations in LoopBack 4 are defined by having a decorator on a designated
+relational property. The definition of the `has many` relation is done by using
+the `hasMany` decorator. The decorator takes in the target model class
+constructor and optionally a custom foreign key to store the relation metadata.
+The decorator logic also designates the relation type and tries to infer the
+foreign key on the target model (or `keyTo` in the relation metadata) as a
+sensible default value which is used in LoopBack 3. It also calls
+`property.array()` to ensure that the type of the property is inferred properly
+as an array of the target model instances. The decorated property name is used
+as the relation name and stored as part of the source model definition's
+relation metadata. The following example shows how to define a `hasMany`
+relation on a source model `Customer`.
+
+{% include code-caption.html content="/src/models/customer.model.ts" %}
+
+```ts
+import {Order} from './order.model.ts';
+import {Entity, property, hasmany} from '@loopback/repository';
+
+export class Customer extends Entity {
+  @property({
+    type: 'number',
+    id: true,
+  })
+  id: number;
+
+  @property({
+    type: 'string',
+  })
+  name: string;
+
+  @hasMany(Order) orders: Order[];
+}
+```
+
+The metadata generated for the above decoration is as follows:
+
+```js
+{
+  type: 2; // denotes hasMany relation
+  keyTo: customerId; // Order model should have this property declared
+}
+```
+
+The property type metadata is also preserved as an array of type `Order` as part
+of the decoration. Another usage of the decorator with a custom foreign key name
+for the above example is as follows:
+
+```ts
+// inside Customer model class
+@hasMany(Order, {keyTo: 'custId'}) orders: Order[];
+```
+
+## Configuring a HasMany relation
+
+Once a `HasMany` relation is defined on the source model, then there are a
+couple of steps involved to configure it and use it. On the source repository,
+the following are required:
+
+- Use [Dependency Injectition](Dependency-injection.md) to inject an instance of
+  the target repository in the constructor of your source repository class.
+- Declare a property with the factory function type
+  `HasManyRepositoryFactory<TargetModel, typeof TargetModel.prototype.id>` on
+  the source repository class.
+- call the `_createHasManyRepositoryFactoryFor` function in the constructor of
+  the source repository class with the relation name (decorated relation propert
+  on the source model) and target repository instance and assign it the property
+  mentioned above.
+
+The following code snippet shows how it would look like:
+
+{% include code-caption.html
+content="/src/repositories/customer.repository.ts.ts" %}
+
+```ts
+{import} {Order} from '../models/order.model.ts';
```

Comment:
Remove the brackets around `import`.
I think this is breaking the markdown rendering